websocket_test.go

```diff
@@ -85,7 +253,44 @@ func TestClient_Send(t *testing.T) {
 	sr := &SimpleResponse{}
 	assert.Equal(t, uint64(0), client.Rx())
 	assert.Equal(t, uint64(0), client.Tx())
-	log.ErrFatal(client.SendProtobuf(servers[0].ServerIdentity, r, sr))
+	require.Nil(t, client.SendProtobuf(servers[0].ServerIdentity, r, sr))
+	assert.Equal(t, sr.Val, 10)
+	assert.NotEqual(t, uint64(0), client.Rx())
+	assert.NotEqual(t, uint64(0), client.Tx())
+	assert.True(t, client.Tx() > client.Rx())
+}
+
+func TestClientTLS_Send(t *testing.T) {
+	cert, key, err := getSelfSignedCertificateAndKey()
+	require.Nil(t, err)
+	CAPool := x509.NewCertPool()
+	CAPool.AppendCertsFromPEM(cert)
+
+	local := NewTCPTestWithTLS(tSuite, cert, key)
+	defer local.CloseAll()
+
+	// register service
+	RegisterNewService(backForthServiceName, func(c *Context) (Service, error) {
+		return &simpleService{
+			ctx: c,
+		}, nil
+	})
+	defer ServiceFactory.Unregister(backForthServiceName)
+
+	// create servers
+	servers, el, _ := local.GenTree(4, false)
+	client := local.NewClient(backForthServiceName)
+	client.tls = true
+	client.trustedCertificates = CAPool
+
+	r := &SimpleRequest{
+		ServerIdentities: el,
+		Val:              10,
+	}
+	sr := &SimpleResponse{}
+	assert.Equal(t, uint64(0), client.Rx())
```

Comment:
I think you need `require` here, else the test will continue, even if this fails.